src/dependabot/verified_commits.ts

```diff
@@ -61,3 +63,54 @@ function warnOtherCommits (): void {
         'any non-Dependabot changes.'
   )
 }
+
+export async function getAlert (name: string, version: string, directory: string, client: InstanceType<typeof GitHub>, context: Context): Promise<dependencyAlert> {
+  const alerts: any = await client.graphql(`
+     {
+       repository(owner: "${context.repo.owner}", name: "${context.repo.repo}") { 
+         vulnerabilityAlerts(first: 100) {
+           nodes {
+             vulnerableManifestFilename
+             vulnerableManifestPath
+             vulnerableRequirements
+             state
+             securityVulnerability { 
+               package { name } 
+             }
+             securityAdvisory { 
+              cvss { score }
+              ghsaId 
+             }
+           }
+         }
+       }
+     }`)
+
+  const nodes = alerts?.repository?.vulnerabilityAlerts?.nodes
+  const found = nodes.find(a => (version === '' || a.vulnerableRequirements === `= ${version}`) &&
+      trimSlashes(a.vulnerableManifestPath) === `${trimSlashes(directory)}/${a.vulnerableManifestFilename}` &&
+      a.securityVulnerability.package.name === name)
+
+  return {
+    alertState: found?.state ?? '',
+    ghsaId: found?.securityAdvisory.ghsaId ?? '',
+    cvss: found?.securityAdvisory.cvss.score ?? 0.0
+  }
+}
+
+export function trimSlashes (value: string): string {
+  return value.replace(/^\/+/, '').replace(/\/+$/, '')
+}
+
+export async function getCompatibility (name: string, oldVersion: string, newVersion: string, ecosystem: string): Promise<number> {
+  const svg = await new Promise<string>((resolve) => {
+    https.get(`https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=${name}&package-manager=${ecosystem}&previous-version=${oldVersion}&new-version=${newVersion}`, res => {
+      let data = ''
+      res.on('data', chunk => { data += chunk.toString('utf8') })
+      res.on('end', () => { resolve(data) })
+    }).on('error', () => { resolve('') })
+  })
+
+  const scoreChunk = svg.match(/<title>compatibility: (?<score>\d+)%<\/title>/m)
+  return scoreChunk?.groups ? parseInt(scoreChunk.groups.score) : 0
```

Comment:
Just double-checking, when we have an incompatible page, e.g. ![](https://dependabot-badges.githubapp.com/badges/compatibility_score.json?dependency-name=licensed&package-manager=bundler&previous-version=3.4.4&new-version=3.5.0)

Line 114 will extract the word `unknown`, which evaluates to 0 here, so that's all good 👍🏻